src/index.ts

```diff
@@ -176,8 +176,9 @@ export interface CreateSelectorFunction<
   ): OutputSelector<
     Selectors,
     Result,
-    (...args: SelectorResultArray<Selectors>) => Result & Keys,
-    GetParamsFromSelectors<Selectors>
+    (...args: SelectorResultArray<Selectors>) => Result,
```

Comment:
I think this was meant to be `((...args: SelectorResultArray<Selectors>) => Result) & Keys`

which would have been closer to working, but it would have put clearCache on resultFunc, which it wasn't actually on, so I needed to pass in the Keys as their own property to be able to add them only to memoizedResultFunc